client/src/reducers/suggestions.js

```diff
@@ -0,0 +1,328 @@
+import {
+    FETCH_CREATE_SUGGESTIONS_PENDING, FETCH_CREATE_SUGGESTIONS_SUCCESS, FETCH_CREATE_SUGGESTIONS_ERROR,
+    FETCH_UPDATE_SUGGESTIONS_PENDING, FETCH_UPDATE_SUGGESTIONS_SUCCESS, FETCH_UPDATE_SUGGESTIONS_ERROR,
+    FETCH_BULK_CREATE_SUGGESTIONS_PENDING, FETCH_BULK_CREATE_SUGGESTIONS_SUCCESS, FETCH_BULK_CREATE_SUGGESTIONS_ERROR,
+    FETCH_CREATE_SUGGESTION_PENDING, FETCH_CREATE_SUGGESTION_SUCCESS, FETCH_CREATE_SUGGESTION_ERROR,
+    FETCH_UPDATE_SUGGESTION_PENDING, FETCH_UPDATE_SUGGESTION_SUCCESS, FETCH_UPDATE_SUGGESTION_ERROR,
+    FETCH_BULK_CREATE_SUGGESTION_PENDING, FETCH_BULK_CREATE_SUGGESTION_SUCCESS, FETCH_BULK_CREATE_SUGGESTION_ERROR,
+    APPROVE_CREATE_SUGGESTION_PENDING, APPROVE_CREATE_SUGGESTION_SUCCESS, APPROVE_CREATE_SUGGESTION_ERROR,
+    REJECT_CREATE_SUGGESTION_PENDING, REJECT_CREATE_SUGGESTION_SUCCESS, REJECT_CREATE_SUGGESTION_ERROR,
+    APPROVE_UPDATE_SUGGESTION_PENDING, APPROVE_UPDATE_SUGGESTION_SUCCESS, APPROVE_UPDATE_SUGGESTION_ERROR,
+    REJECT_UPDATE_SUGGESTION_PENDING, REJECT_UPDATE_SUGGESTION_SUCCESS, REJECT_UPDATE_SUGGESTION_ERROR,
+    APPROVE_BULK_CREATE_SUGGESTION_PENDING, APPROVE_BULK_CREATE_SUGGESTION_SUCCESS, APPROVE_BULK_CREATE_SUGGESTION_ERROR,
+    REJECT_BULK_CREATE_SUGGESTION_PENDING, REJECT_BULK_CREATE_SUGGESTION_SUCCESS, REJECT_BULK_CREATE_SUGGESTION_ERROR
+} from '../constants';
+import { LOCATION_CHANGE } from 'connected-react-router';
+
+const initialState = {
+    pending: false,
+    result: null,
+    error: null
+};
+
+export function fetchCreateSuggestions(state = initialState, action) {
+    switch (action.type) {
+        case FETCH_CREATE_SUGGESTIONS_PENDING:
+            return {
+                ...state,
+                pending: true
+            };
+        case FETCH_CREATE_SUGGESTIONS_SUCCESS:
+            return {
+                ...state,
+                pending: false,
+                result: action.payload
+            };
+        case FETCH_CREATE_SUGGESTIONS_ERROR:
+            return {
+                ...state,
+                pending: false,
+                error: action.error
+            };
+        default:
+            return state;
+    }
+}
+
```

Comment:
![image](https://user-images.githubusercontent.com/16584007/79133807-a4a54c80-7d7a-11ea-8f84-2b6803c897c8.png)

Reply:
It does that to give us the most flexibility. There are many places where we do `dispatch(success(actions.blah, {payload: result}));` instead of `dispatch(success(actions.blah, result));`, which will give you the result you're looking for I believe (where `state.blah.payload` exists in `mapStateToProps`)

Reply:
Ok, changed them all to use `basicReducer` besides the one for approve Bulk-Create Suggestions since it utilizes `progress` and `basicReducer` doesn't account for that.